Since Developer Studio 12.5 new atomics libraries are in place.
Following fix is based on MySQL 5.7 change.

diff --git a/cmake/build_configurations/compiler_options.cmake b/cmake/build_configurations/compiler_options.cmake
index 80ae8f2..8ff8056 100644
--- a/cmake/build_configurations/compiler_options.cmake
+++ b/cmake/build_configurations/compiler_options.cmake
@@ -101,8 +101,8 @@ IF(UNIX)
         ENDIF()
       ELSE() 
         # Assume !x86 is SPARC
-        SET(COMMON_C_FLAGS                 "-g -xstrconst -mt")
-        SET(COMMON_CXX_FLAGS               "-g0 -mt")
+        SET(COMMON_C_FLAGS                 "-g -xstrconst -mt -xatomic=gcc")
+        SET(COMMON_CXX_FLAGS               "-g0 -mt -xatomic=gcc")
         IF(32BIT)
           SET(COMMON_C_FLAGS               "${COMMON_C_FLAGS} -xarch=sparc")
           SET(COMMON_CXX_FLAGS             "${COMMON_CXX_FLAGS} -xarch=sparc")
